server/catgenome/src/main/java/com/epam/catgenome/manager/externaldb/homolog/HomologManager.java

```diff
@@ -0,0 +1,332 @@
+ * SOFTWARE.
+ */
+package com.epam.catgenome.manager.externaldb.homolog;
+
+import com.epam.catgenome.component.MessageCode;
+import com.epam.catgenome.dao.homolog.HomologDatabaseDao;
+import com.epam.catgenome.dao.homolog.HomologGeneAliasDao;
+import com.epam.catgenome.dao.homolog.HomologGeneDescDao;
+import com.epam.catgenome.dao.homolog.HomologGeneDomainDao;
+import com.epam.catgenome.dao.homolog.HomologGroupDao;
+import com.epam.catgenome.dao.homolog.HomologGroupGeneDao;
+import com.epam.catgenome.entity.externaldb.homolog.HomologDatabase;
+import com.epam.catgenome.entity.externaldb.homolog.HomologGroup;
+import com.epam.catgenome.entity.externaldb.homolog.HomologGroupGene;
+import com.epam.catgenome.entity.externaldb.homolog.HomologType;
+import com.epam.catgenome.entity.externaldb.homologene.Alias;
+import com.epam.catgenome.entity.externaldb.homologene.Domain;
+import com.epam.catgenome.entity.externaldb.homologene.Gene;
+import com.epam.catgenome.manager.externaldb.SearchResult;
+import com.epam.catgenome.util.db.Filter;
+import com.epam.catgenome.util.db.PagingInfo;
+import com.epam.catgenome.util.db.QueryParameters;
+import lombok.Builder;
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+import lombok.Setter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.http.util.TextUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+import org.springframework.util.Assert;
+import org.springframework.util.CollectionUtils;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static com.epam.catgenome.component.MessageHelper.getMessage;
+import static com.epam.catgenome.util.Utils.DEFAULT_PAGE_SIZE;
+import static java.util.stream.Collectors.groupingBy;
+import static org.apache.commons.lang3.StringUtils.join;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class HomologManager {
+
+    private static final String FIELDS_LINE_DELIMITER = "\t";
+
+    @Value("${homologene.index.directory}")
+    private String indexDirectory;
+
+    @Autowired
+    HomologGroupDao homologGroupDao;
+
+    @Autowired
+    HomologGroupGeneDao homologGroupGeneDao;
+
+    @Autowired
+    HomologDatabaseDao homologDatabaseDao;
+
+    @Autowired
+    HomologGeneDescDao homologGeneDescDao;
+
+    @Autowired
+    HomologGeneAliasDao homologGeneAliasDao;
+
+    @Autowired
+    HomologGeneDomainDao homologGeneDomainDao;
+
+    public SearchResult<HomologGroup> searchHomolog(final HomologSearchRequest request)
```

Comment:
Could you please move public methods to the top of the class and then private ones?